integration-tests/bats/keyless.bats

```diff
@@ -984,3 +984,191 @@ SQL
     [[ "$output" =~ "i" ]] || false
     [[ "$output" =~ "1" ]] || false
 }
+
+@test "keyless: insert into keyless table with unique index" {
+    dolt sql -q "CREATE TABLE mytable (pk int UNIQUE)";
+
+    run dolt sql -q "INSERT INTO mytable values (1),(2),(3),(4)"
+    [ $status -eq 0 ]
+
+    run dolt sql -r csv -q "SELECT * FROM mytable order by pk"
+    [ $status -eq 0 ]
+    [[ "${lines[1]}" = "1" ]] || false
+    [[ "${lines[2]}" = "2" ]] || false
+    [[ "${lines[3]}" = "3" ]] || false
+    [[ "${lines[4]}" = "4" ]] || false
+
+    run dolt sql -q "INSERT INTO mytable VALUES (1)"
+    [ $status -eq 1 ]
+    [[ "$output" =~ "duplicate unique key given: [1]" ]] || false
+
+    # Make sure nothing in the faulty transaction was persisted.
+    run dolt sql -q "INSERT INTO mytable VALUES (500000), (5000001), (3)"
+    [ $status -eq 1 ]
+    [[ "$output" =~ "duplicate unique key given: [3]" ]] || false
+
+    run dolt sql -r csv -q "SELECT count(*) FROM mytable where pk = 5"
+    [ $status -eq 0 ]
+    [[ "${lines[1]}" = "0" ]] || false
+
+    run dolt sql -r csv -q "SELECT count(*) FROM mytable"
+    [ $status -eq 0 ]
+    [[ "${lines[1]}" = "4" ]] || false
+
+    run dolt index cat mytable pk -r csv
+    [ $status -eq 0 ]
+    [[ "${lines[0]}" = "pk" ]] || false
+    [[ "${lines[1]}" = "1" ]] || false
+    [[ "${lines[2]}" = "2" ]] || false
+    [[ "${lines[3]}" = "3" ]] || false
+    [[ "${lines[4]}" = "4" ]] || false
+    [[ "${#lines[@]}" = "5" ]] || false
+}
+
+@test "keyless: insert into keyless table with unique index and auto increment" {
+    dolt sql -q "CREATE TABLE gis (pk INT UNIQUE NOT NULL AUTO_INCREMENT, shape GEOMETRY NOT NULL)"
+    dolt sql -q "INSERT INTO gis VALUES (1, POINT(1,1))"
+
+    run dolt sql -q "INSERT INTO gis VALUES (1, POINT(1,1))"
+    [ $status -eq 1 ]
+    [[ "$output" =~ "duplicate unique key given: [1]" ]] || false
+
+    run dolt sql -r csv -q "SELECT count(*) FROM gis where pk = 1"
+    [ $status -eq 0 ]
+    [[ "${lines[1]}" = "1" ]] || false
+
+    run dolt sql -q "INSERT INTO gis VALUES (NULL, POINT(1,1))"
```

Comment:
@max-hoffman This is standard Auto Increment logic. You can put in NULL and it will get resolved with the next auto increment value.